launcher/ui/widgets/ThemeCustomizationWidget.h

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (C) 2022 Tayou <tayou@gmx.net>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <QWidget>
+#include <translations/TranslationsModel.h>
+
+enum ThemeFields { 
+    NONE    = 0b0000,
+    ICONS   = 0b0001,
+    WIDGETS = 0b0010,
+    CAT     = 0b0100
+};
+
+namespace Ui {
+class ThemeCustomizationWidget;
+}
+
+class ThemeCustomizationWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    explicit ThemeCustomizationWidget(QWidget *parent = nullptr);
+    ~ThemeCustomizationWidget();
+
+    void showFeatures(ThemeFields features);
+
+    void applySettings();
+
+    void loadSettings();
+    void retranslate();
+    
+    Ui::ThemeCustomizationWidget *ui;
+
+private slots:
+    void applyIconTheme(int index);
+    void applyWidgetTheme(int index);
+    void applyCatTheme(int index);
+
+signals:
+    int currentIconThemeChanged(int index);
+    int currentWidgetThemeChanged(int index);
+    int currentCatChanged(int index);
+
+private:
+
+    QStringList m_iconThemeOptions{ "pe_colored", "pe_light", "pe_dark", "pe_blue", "breeze_light", "breeze_dark", "OSX", "iOS", "flat", "flat_white", "multimc", "custom" };
+    QStringList m_catOptions{ "kitteh", "rory", "rory-flat" };

```

Comment:
it seems like you're correlating the index in these string lists with the indexes of the elements in the .ui, right? wouldn't it make more sense to instead have these lists contain both the "slug" of the options (as they currently do), as well as the human-readable names of those, and just initialize the ui list in the ctor? something like
```cpp
QList<std::pair<QString, QString>> m_iconThemeOptions {
{ QStringLiteral("pe_colored"), QObject::tr("Simple (Colored Icons)") },
{ QStringLiteral("pe_light"),   QObject::tr("Simple (Light Icons)")   },
...
}
```

while it would be a bit more complex in here, it eliminates the cross indexing that currently exists between the elements here and the ones in the .ui, so it would be harder for someone to accidentally break the logic by changing only one of them. With that, you could also mirror the theme logic in `loadSettings` to the icon themes as well, which would make things a bit cleaner imo :)

Reply:
yes, that would make sense
Maybe even using a QMap or QHash instead?

Reply:
it would work, yes, but if all you'll do with it is traverse all the elements, there's no need for the extra overhead of those. It isn't that big of a deal either way, so do as you wish :)